### PR TITLE
Allow configuring GRPC max connection age and max connection age grace

### PR DESCRIPTION
**Pull request by siweili11**
Add ability to configure GRPC max connection age and max connection age grace. The definitions of the two parameters can be found in:

https://grpc.io/docs/guides/keepalive/#keepalive-configuration-specification

https://grpc.github.io/grpc/cpp/group__grpc__arg__keys.html#gaf4574abe94c339c6f21163bca6e7b6b7

Same as the initial GRPC proposal: https://github.com/grpc/proposal/blob/master/A9-server-side-conn-mgt.md#proposal, the reason why we need to configure the two parameters is to spread load more quickly, by forcing the client to periodically reconnect to a L4 LB.

### CODE REVIEW

**Comment by jbkyang-nvi** [comment]
Hi! Thanks for your contribution. Can you fill in the [CLA](https://github.com/triton-inference-server/server/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) for this?

**Comment by siweili11** [comment]
Hi, thanks for the guidance! I have reformatted my changes and rebased. Please take a look again when you have a chance.

I'll fill in the CLA before this code change is merged

File: src/grpc/grpc_server.h
Comment (jbkyang-nvi) [inline src/grpc/grpc_server.h:78]: actually per convention can you just set this to 0?

File: src/grpc/grpc_server.cc
Comment (jbkyang-nvi) [inline src/grpc/grpc_server.cc:2369]: can you just check if the parameter is populated and only `AddChannelArgument` if it is nonzero?

**Review by jbkyang-nvi — CHANGES_REQUESTED** [review]

File: src/grpc/grpc_server.h
Comment (siweili11) [reply src/grpc/grpc_server.h:78]: If I understand the code correctly, the value inside {} is the default value of the option if the flag is not set. So I want the default values here to be consistent with GRPC default values, see https://grpc.io/docs/guides/keepalive/#keepalive-configuration-specification. 

Basically GRPC by default tend to re-use existing connections as possible as it can to reduce network overhead, I think we should preserve this behavior in Triton unless the flags are set.

Please let me know if my understanding is incorrect.

File: src/grpc/grpc_server.cc
Comment (siweili11) [reply src/grpc/grpc_server.cc:2369]: Sorry I'm still new to this. Can you please provide more details on how to check whether the parameter is populated?

Comment (siweili11) [reply src/grpc/grpc_server.cc:2369]: Or do you mean I should make code changes such that AddChannelArgument will only be invoked when the parameter is set by the user with a non-default value?

File: src/grpc/grpc_server.h
Comment (jbkyang-nvi) [reply src/grpc/grpc_server.h:78]: Sorry for the confusion here. I was trying to say instead of specifying the max connection age, just don't specify `max_connection_age_ms_` or `max_connection_age_grace_ms_` upon initialization of the `KeepAliveOptions` struct and never pass these parameters to the server if the values are 0.

Comment (siweili11) [reply src/grpc/grpc_server.h]: Thanks for the explanation, I have updated my code change according to comments

File: src/grpc/grpc_server.h
Comment (jbkyang-nvi) [inline src/grpc/grpc_server.h:77]: explicitly initialize to 0 here

**Review by jbkyang-nvi — COMMENTED** [review]
Looks good! If you make the changes and file the CLA, I will kick off a pipeline run in our internal CI, we should be good to go.

**Comment by siweili11** [comment]
Hi, my company has sent an email to triton-cla@nvidia.com requesting to add me to my company CLA. Can you please approve when you have a chance?

**Comment by jbkyang-nvi** [comment]
@siweili11  can you *explicitly* initialize to 0?

**Comment by siweili11** [comment]
Thanks. I have updated my code change to initialize as 0

**Review by jbkyang-nvi — APPROVED** [review]
Thanks for your contribution!

**Comment by siweili11** [comment]
Thanks for merging! May I know when this change can be included in a released image?

**Comment by jbkyang-nvi** [comment]
This will be available in the 24.01 image (our freeze passed a week ago)

**Comment by jbkyang-nvi** [comment]
> Hi，how can i set these params when start a trion server?

you should be able to set `--grpc-max-connection-age` and `--grpc-max-connection-age-grace` in the command line when you run tritonserver. Details can be found if you do `tritonserver --help`

**Pull request merged**
